### PR TITLE
Add section on difference array

### PR DESCRIPTION
**Pull request by IDON-TEXIST**
_Place an "x" in the corresponding checkbox if it is done or does not apply to this pull request._

- [ ] I have tested my code.
- [ ] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [ ] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions).
  - I understand that if it is clear that I have not attempted to follow these conventions, my PR will be closed.
  - If changes are requested, I will re-request a review after addressing them.
- [ ] I have linked this PR to any issues that it closes.

Not very proud of the wording. I'll come up with something better someday.

### CODE REVIEW

File: content/3_Silver/Prefix_Sums.mdx [hunk 1]
```diff
@@ -293,6 +293,19 @@ for i in range(Q):
 
 </LanguageSection>
 
+## Difference Array
+
+<FocusProblem problem="haybale-stacking" />
+
+<Spoiler title = "Hint">
+What if you did prefix sums, but in reverse?
```
Comment (envyaims) [inline content/3_Silver/Prefix_Sums.mdx:301]: what

File: content/3_Silver/Prefix_Sums.mdx [hunk 2]
```diff
@@ -293,6 +293,19 @@ for i in range(Q):
...
+</Spoiler>
+
+<Spoiler title = "Solution">
+Consider the array $[0, 1, 0, 0, 0]$. Running prefix sums on this gives us $[0, 1, 1, 1, 1]$. Using prefix sums, we can implicitly add to a suffix by changing one number.
```
Comment (envyaims) [inline content/3_Silver/Prefix_Sums.mdx:305]: include a better example with 1s as well as -1s.
state that this is equivalent to a range addition

**Review by envyaims — CHANGES_REQUESTED** [review]

File: content/3_Silver/Prefix_Sums.mdx (hunk 1 above)
Comment (ryanchou-dev) [reply content/3_Silver/Prefix_Sums.mdx:301]: wait yeah this doesn't make a lot of sense 😭

**Comment by IDON-TEXIST** [comment]
Eh.. this feels like a worse and worse idea the more I think about it. I'll let it simmer.

**Pull request closed without merging**